docs/examples/writing_drivers/Creating-Instrument-Drivers.ipynb

```diff
@@ -83,7 +83,15 @@
     "  \n",
     "Golden rule: if a `Parameter` is settable, it must always accept its own output as input.\n",
     "\n",
-    "In most cases you will probably be adding parameters via the `add_parameter` method on the instrument class as shown in the example below. \n",
+    "There are two different ways of adding parameters to instruments. They are almost equivalent but comes with some trade-offs. We will show both below.\n",
+    "You may either declare the parameter as an attribute directly on the instrument or add it via the via the `add_parameter` method on the instrument class.\n",
+    "\n",
+    "Declaring a parameter as an attribute directly on the instrument enables Sphinx, IDEs such as VSCode and static tools such as Mypy to work more fluently with \n",
+    "the parameter than if it is created via `add_parameter` however you must take care to remember to pass `instrument=self` to the parameter such that the\n",
+    "parameter will know which instrument it belongs to. \n",
+    "Instrument.add_parameter is better suited for when you want to dynamically or programmatically add a parameter to an instrument. For historical reasons most \n",
+    "instruments currently use `add_parameter`.\n",
+    "\n",
```

Comment:
Perhaps this notebook should document that parameters must pass kwargs to the baseclass